docs/source/Examples/lambci.rst

```diff
@@ -0,0 +1,312 @@
+Using a custom runtime
+*******************************************************************************
+
+A custom runtime replaces the runtime environment provided by the FaaS provider
+with an own runtime package. This package has to be uploaded as a layer to the
+FaaS provider.
+
+To use a custom runtime, specify ``provided`` as the runtime name for the
+Docker command.
+
+::
+
+	$ ./lambci.sh ~/lambci provided lambda_function.lambda_handler
+
+The lambda container now expects the custom lambda runtime in the ``runtime``
+directory. For this, create a layer that contains the runtime code and configure
+it in the configuration.
+
+::
+
+	default-provider : "lambci"
+
+	providers :
+	  lambci :
+	    faas : "lambciLambda"
+
+	service :
+	 faas :
+	    lambciLambda:
+	      # path to the lambci directory
+	      directory : '~/lambci'
+	      # address of lambci server API
+	      address : 'localhost:9001'
+	      # runtime configuration
+	      runtimes :
+	        # with python requests package
+	        custom-runtime :
+	          # list of additional layers 
+	          layers :
+	            - 'custom-python'
+	            - 'requests'
+
+The custom runtime can then be specified at function creation. In the example
+above, SRK will copy the contents of the ``custom-python`` directory (the
+custom runtime) and the ``requests`` layer to the ``runtime`` directory so that
+the LambCI ``provided`` container finds it in ``/opt``.
+
+::
+
+	$ ./srk --config configs/local-srk.yaml function create -s examples/echo -r custom-runtime
+
+
+*******************************************************************************
+Scripting LambCI Startup
+*******************************************************************************
+
+LambCI provides docker images for various runtimes out of the box, but can also
+use a custom runtime. To inject the lambda function into the container, the
+``/var/task`` and ``/opt`` directories are mounted to local directories by the
+``docker run`` command. To be compatible with the SRK these directories need to
+be inside the configured LambCI home directory and have the names ``task`` for
+the lambda function and ``runtime`` for a custom runtime or additional layer
+files.
+
+Instead of running the lambda function immediately, SRK uses the LambCI-provided
+webserver with an invocation API to execute the lambda function. Therefore the
+port of the webserver has to be exposed by the ``docker run`` command.
+
+It is also possible to inject environment variables via the ``--env-file``
+parameter of ``docker run``.
+
+Updates to the function, the runtime files or the environment file require a
+restart of the container. The helper program ``entr`` can be used to automate
+this. It can be installed via ``apt install entr`` (Ubuntu),
+``yum install entr`` (Amazon Linux 2) or ``brew install entr`` on MacOS X.
+
+Please see the following shell script as a loader for a LambCI lambda
+container.
+
+::
```

Comment:
we might consider just adding this to the tools/ directory instead of having users create it themselves.

Reply:
That makes sense, and as I recall Leonhard did that originally. However, I don't believe this script is robust enough. It relies on the `entr` command which isn't always present on all systems (not on Mac by default). We will come up with a more robust solution eventually.